### PR TITLE
add standard range sharding algorithm and refactor related code

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #5280.

Changes proposed in this pull request:
- add standard range sharding algorithm and refactor related code

### CODE REVIEW

File: sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/StandardRangeShardingAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    @Override
+    public void initProperties() {
+        Preconditions.checkNotNull(properties.get(PARTITION_LOWER), "Standard range sharding algorithm partition lower cannot be null.");
+        Preconditions.checkNotNull(properties.get(PARTITION_UPPER), "Standard range sharding algorithm partition upper cannot be null.");
+        Preconditions.checkNotNull(properties.get(PARTITION_VOLUME), "Standard range sharding algorithm partition volume cannot be null.");
+        long lower = Long.parseLong(properties.get(PARTITION_LOWER).toString());
+        long upper = Long.parseLong(properties.get(PARTITION_UPPER).toString());
+        long volume = Long.parseLong(properties.get(PARTITION_VOLUME).toString());
+        Preconditions.checkArgument(upper - lower >= volume, "Standard range sharding algorithm partition range can not be smaller than volume.");
+
```
Comment (tristaZero) [inline sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/StandardRangeShardingAlgorithm.java:87]: A redundant blank line is expected to remove.

File: sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/CustomRangeShardingAlgorithm.java [hunk 2]
```diff
@@ -51,61 +49,39 @@
  * The sharding values will be divided into different partition by its value.
  * </p>
  */
-public final class RangeShardingAlgorithm implements StandardShardingAlgorithm<Long> {
+public final class CustomRangeShardingAlgorithm extends AbstractRangeShardingAlgorithm {
 
     private static final String PARTITION_RANGES = "partition.ranges";
 
     private Map<Integer, Range<Long>> partitionRangeMap;
```
Comment (tristaZero) [inline sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/CustomRangeShardingAlgorithm.java:56]: Hi, do you think it is appropriate to move `partitionRangeMap` to `AbstractRangeShardingAlgorithm `?

**Comment by tristaZero** [comment]
@strongduanmu 
Superb! I could pick up many highlights from your design. Just two comments need to look at. 😀

File: sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/StandardRangeShardingAlgorithm.java (hunk 1 above)
Comment (strongduanmu) [reply sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/StandardRangeShardingAlgorithm.java:87]: > A redundant blank line is expected to remove.

Ok, I will modify it.

File: sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/CustomRangeShardingAlgorithm.java (hunk 2 above)
Comment (strongduanmu) [reply sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/CustomRangeShardingAlgorithm.java:56]: > Hi, do you think it is appropriate to move `partitionRangeMap` to `AbstractRangeShardingAlgorithm `?
@tristaZero Before I submit the pr, I hava already tried to put the partitionRangeMap in the AbstractRangeShardingAlgorithm class, decorated with a protected identifier, but the CheckStyle detection prompt that it should be defined as a private type, and accessed through the getter/setter method. I think it is not elegant if we use the getter/setter method in the subclass to operate the partitionRangeMap.

Comment (tristaZero) [reply sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/range/CustomRangeShardingAlgorithm.java:56]: Got your idea, it is not a big deal, and thanks for the clarification. :)

**Comment by tristaZero** [comment]
Got your idea; it is not a big deal. Moreover, thanks for your much effort and deliberation for sharding algorithm. Great job!
@strongduanmu

**Pull request merged**